### DISCUSSION TITLE
Black & Litterman Analyst Views & Group Constraints

### DISCUSSION THREAD

**Discussion by devoneup** [skfolio/skfolio#93] [General]
Hi, 

I was wondering if there was a way to implement constraints on groups whose composition is varying over time? In a similar vein, analysts views do change over time. Is there a way to take that into account in Black & Litterman?

Thanks!

**Comment by HugoDelatte** [comment]
Hello,

Yes it's possible, you can apply a new group constrain or analys view per period.
You just need to call each optimization `fit` and `predict` yourself outside the `cross_val_predict` function.
